hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -119,32 +117,26 @@ private boolean skipUpdateStatistics() {
     }
 
     private void handleExecuteResponse(
-        ClientConnection connection,
         SqlClientResult res,
-        ClientMessage message,
-        Throwable error
+        ClientMessage message
     ) {
-        if (error != null) {
-            res.onExecuteError(rethrow(error, connection));
-
-            return;
-        }
-
         SqlExecuteCodec.ResponseParameters response = SqlExecuteCodec.decodeResponse(message);
-
-        HazelcastSqlException responseError = handleResponseError(response.error);
-
-        if (responseError != null) {
-            res.onExecuteError(responseError);
-
-            return;
+        SqlError sqlError = response.error;
+        if (sqlError != null) {
+            throw new HazelcastSqlException(
+                    sqlError.getOriginatingMemberId(),
+                    sqlError.getCode(),
+                    sqlError.getMessage(),
+                    null,
+                    sqlError.getSuggestion()
+            );
```

Comment:
We've lost the `res.onExecuteError` call - it sets the state for the client cursor and for future calls.
```suggestion
            HazelcastSqlException responseError = handleResponseError(sqlError);
            res.onExecuteError(responseError);
            throw responseError;
```

Reply:
However we will call it twice if we have it here.

Reply:
Ok, missed that.